en/guide/define_xml_element.md

```diff
@@ -0,0 +1,336 @@
+   - Add additional `<include> </include>` elements if you want to import additional files/dialects.
+     > **Note** Includes in nested files are ignored.
+1. Update the version:
+   - if the dialect is not based on **common.xml** uncomment the `<version>6</version>` line and use whatever version you like.
+       > **Note** The version specified in the current file is used by default. If it is not present then the version in included files is used. Most dialects inherit the version from **common.xml**.
+1. Update the `<dialect>8</dialect>` line to replace `8` with the next-largest unused dialect number (based on the other files in the folder).
+1. Optionally remove the `enums` or `messages` sections if you don't plan on declaring any of these elements.
+1. Add enums or messages as described in the following sections.
+1. Save the file, and create a PR to push it back to the **mavlink/mavlink** project repo.
+
+
+## Messages
+
+Messages are used to send data between MAVLink systems (including commands, information and acknowledgments).
+
+Every message has mandatory `id` and `name` attributes.
+[Serialised packets](../guide/serialization.md#packet_format) include the `id` in the [message id](../guide/serialization.md#v1_msgid) section and an encoded form of the message data within the [payload](../guide/serialization.md#v1_payload) section. 
+The `name` is typically used by generators to name methods for encoding and decoding the specific message type.
+When a message is received the MAVLink library extracts the message id to determine the specific message, and uses that to find the appropriately named method for decoding the payload.
+
+A typical message ([SAFETY_SET_ALLOWED_AREA](../messages/common.md#SAFETY_SET_ALLOWED_AREA)) is shown below:
+
+```xml
+    <message id="54" name="SAFETY_SET_ALLOWED_AREA">
+      <description>Set a safety zone (volume), which is defined by two corners of a cube. This message can be used to tell the MAV which setpoints/waypoints to accept and which to reject. Safety areas are often enforced by national or competition regulations.</description>
+      <field type="uint8_t" name="target_system">System ID</field>
+      <field type="uint8_t" name="target_component">Component ID</field>
+      <field type="uint8_t" name="frame" enum="MAV_FRAME">Coordinate frame. Can be either global, GPS, right-handed with Z axis up or local, right handed, Z axis down.</field>
+      <field type="float" name="p1x" units="m">x position 1 / Latitude 1</field>
+      <field type="float" name="p1y" units="m">y position 1 / Longitude 1</field>
+      <field type="float" name="p1z" units="m">z position 1 / Altitude 1</field>
+      <field type="float" name="p2x" units="m">x position 2 / Latitude 2</field>
+      <field type="float" name="p2y" units="m">y position 2 / Longitude 2</field>
+      <field type="float" name="p2z" units="m">z position 2 / Altitude 2</field>
+    </message>
+```
+
+
+### Creating a Message
```

Comment:
@amilcarlucas Just FYI, this is still work in progress, but I have updated this by inferring the answers to some of my questions. I'll continue with enums tomorrow.

Basically you can infer that there are allocated ranges for message ids in MAVLink 2. I have suggested creating an issue to request a new range.

I also tested what mavgen reports errors against.

Reply:
Nice !